src/tools/prefets/notificationsNouvellesDemandesStructures.ts

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env node
+/* eslint-disable no-await-in-loop */
+
+// Lancement de ce script : ts-node src/tools/prefets/notificationsNouvellesDemandesStructures.ts
+
+import execute from '../utils';
+import service from '../../helpers/services';
+import { informationNouvelleCandidatureStructure } from '../../emails';
+import { IStructures } from '../../ts/interfaces/db.interfaces';
+
+execute(__filename, async ({ app, mailer, logger, exit }) => {
+  const dateMoins1Jours = new Date();
+  dateMoins1Jours.setDate(dateMoins1Jours.getDate() - 1);
+  const structures: IStructures[] = await app
+    .service(service.structures)
+    .Model.find(
+      {
+        statut: 'CREEE',
+        coordinateurCandidature: false,
+        createdAt: { $gte: dateMoins1Jours, $lte: new Date() },
+        codeDepartement: { $ne: '00' },
+      },
+      {
+        codeDepartement: 1,
+        nom: 1,
+      },
+    );
+  if (structures.length === 0) {
+    exit();
+    return;
+  }
+  const departements = [
+    ...new Set(structures.map((structure) => structure.codeDepartement)),
+  ];
+  const prefets = await app.service(service.users).Model.find(
+    {
+      departement: { $in: departements },
+    },
+    {
+      _id: 0,
+      name: 1,
+      departement: 1,
+    },
+  );
+  const prefetsWithStructure = structures.flatMap((structure) =>
+    prefets
+      .filter((prefet) => prefet.departement === structure.codeDepartement)
+      .map((prefet) => ({ ...prefet._doc, structure })),
+  );
+  const messageAvisCandidaturePosteStructure =
+    informationNouvelleCandidatureStructure(app, mailer);
+  const promises: Promise<void>[] = [];
+  let count = 0;
+  prefetsWithStructure.forEach(async (prefetWithStructure) => {
```

Comment:
peut etre passer en for of avec bounce. Brevo peut râler si trop de mail one shot